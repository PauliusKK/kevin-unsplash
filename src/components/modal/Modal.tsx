import React from 'react';
import ReactModal from 'react-modal';
import classnames from 'classnames'
import { useSelector, useDispatch } from 'react-redux';
import { addLiked, deleteLiked, setModalOpen } from '../../actions/rootActions';

import CloseIcon from '../../assets/images/close.svg';
import DefaultAuthorIcon from '../../assets/images/default-author.svg';
import SmallLikedIcon from '../../assets/images/small-heart.svg';

import './scss/modal.scss';

export interface Photo {
  id: string;
  source: string;
  alt: string;
  fullSource: string;
  liked: boolean;
  description: string;
  name: string;
  profilePicture: string;
  user: {};
}

export interface PhotoProps {
  photo: Photo;
}

export const Modal = ({ photo }: PhotoProps) => {
  const isModalOpen = useSelector((state: any) => state.isModalOpen)
  const dispatch = useDispatch();

  return (
    <ReactModal
      isOpen={isModalOpen}
      onRequestClose={() => dispatch(setModalOpen(false))}
      style={{ overlay: { display: 'flex', zIndex: 999, backgroundColor: 'rgba(43, 43, 43, 0.7)' } }}
      ariaHideApp={false}
      className="modal"
    >
      <div className="image-box">
        {photo.fullSource && photo.alt && (
          <img src={photo.fullSource} alt={photo.alt} />
        )}
      </div>

      <div className="description-box">
        <div className="cta-box">
          <button
            className={classnames('cta', photo.liked && 'liked')}
            onClick={() => {
              photo.liked ? dispatch(deleteLiked(photo)) : dispatch(addLiked(photo))
            }}
          >
            <img src={SmallLikedIcon} alt={'icon'} />
            <span>{photo.liked ? 'Unlike' : 'Like'}</span>
          </button>

          <button className="modal-close" onClick={() => dispatch(setModalOpen(false))}>
            <img src={CloseIcon} alt="Close" />
          </button>
        </div>

        {photo.description && <h2 className="description">{photo.description}</h2>}

        {photo.name && (
          <div className="author-info">
            <img
              src={photo.profilePicture ? photo.profilePicture : DefaultAuthorIcon}
              alt={photo.name}
              className="author-photo"
            />
            <span className="author-name">{photo.name}</span>
          </div>
        )}

        <div className="specifications">
          {photo.user && Object.entries(photo.user).map(([key, value]: any) => (
            <div className="specification">
              <p>{key}</p>
              <h5>{value}</h5>
            </div>
          ))}
        </div>
      </div>
    </ReactModal>
  )
};