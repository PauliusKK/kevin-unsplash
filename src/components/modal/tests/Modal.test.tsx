import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';
import { Modal } from '../Modal';

const mockStore = configureMockStore([thunk]);

const samplePhoto = {
  alt: "person in brown long sleeve shirt using silver Surface",
  description: "person in brown long sleeve shirt using silver Surface",
  fullSource: "https://images.unsplash.com/photo-1612832021455-245704c6755a?crop=entropy&cs=srgb&fm=jpg&ixid=MXwyMTA4Mzd8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=85",
  id: "TQmnNr9odYs",
  liked: true,
  name: "Surface",
  profilePicture: "https://images.unsplash.com/profile-1587651800415-20eed2ec0209image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=32&w=32",
  regularSource: "https://images.unsplash.com/photo-1612832021455-245704c6755a?crop=entropy&cs=tinysrgb&fit=max&fm=jpg&ixid=MXwyMTA4Mzd8MXwxfGFsbHwxfHx8fHx8Mnw&ixlib=rb-1.2.1&q=80&w=1080",
  user: {
    allPhotos: 84,
    collections: 0,
    instagram: "@surface",
    likes: 0,
    twitter: "@surface",
    username: "surface"
  }
};

describe('Modal test suite', () => {
  describe('ComponentModal', () => {
    test('renders correctly', () => {
      const store = mockStore({
        photo: samplePhoto,
      });
      const { asFragment } = render(
        <Provider store={store}>
          <Router>
            <Modal photo={store.photo || samplePhoto} />
          </Router>
        </Provider>
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
})