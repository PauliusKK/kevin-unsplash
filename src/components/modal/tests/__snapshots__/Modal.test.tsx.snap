// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal test suite ComponentModal renders correctly 1`] = `<DocumentFragment />`;
